app/integration_action.go

```diff
@@ -207,6 +207,11 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 // Perform an HTTP POST request to an integration's action endpoint.
 // Caller must consume and close returned http.Response as necessary.
 func (a *App) DoActionRequest(rawURL string, body []byte) (*http.Response, *model.AppError) {
+	siteURL, _ := url.Parse(*a.Config().ServiceSettings.SiteURL)
+	if strings.HasPrefix(rawURL, "/plugins/") {
+		rawURL = siteURL.String() + rawURL
```

Comment:
Minor but could use https://golang.org/pkg/path/#Join to do this more cleanly in case their are trailing spaces or extra slashes

Reply:
I've actually thought about using that but it would require rebuilding the full URL from its components (scheme, hostname, etc) because we are not dealing with just paths there.
If that makes sense I can update.

Reply:
0/5 it's not a huge deal

Reply:
[url.Parse](https://golang.org/pkg/net/url/#Parse) supports both relative and absolute URLs, leaving `Host` empty in the former case. Perhaps that plus `path.Join` would alleviate most of the URL rebuild pain? One benefit would be that it also parses relative URLs without a leading slash  -- making the interface just a little more friendly to plugin developers.

Reply:
@lieut-data thank you for your feedback
It took some thinking but I've just pushed an update that handles the relative URL case more robustly. I've also made some slight changes to the flow of things. I think it makes more sense failing as soon as possible if the input URL is not valid and avoiding creating the HTTP request.